core/controllers/suggestion.py

```diff
@@ -761,15 +765,27 @@ def get(self, target_type: str, suggestion_type: str) -> None:
         suggestions: Sequence[suggestion_registry.BaseSuggestion] = []
         next_offset = 0
         if suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT:
-            reviewable_suggestions, next_offset = (
-                suggestion_services
-                .get_reviewable_translation_suggestions_by_offset(
-                    self.user_id, exp_ids, limit, offset, sort_key))
+            reviewable_suggestions: List[
+                suggestion_registry.SuggestionTranslateContent] = []
+            if (exp_ids and len(exp_ids) == 1 and limit is None):
+                reviewable_suggestions, next_offset = (
+                    suggestion_services
+                    .get_reviewable_translation_suggestions_for_single_exp(
+                        self.user_id, exp_ids[0]))
+            else:
+                reviewable_suggestions, next_offset = (
```

Comment:
Probably you need a TODO here to remove the need for the else branch, right? (corresponding to the same issue you filed before)

Reply:
I don't think so. The if branch fetches the translation suggestions/cards for a reviewable lesson. Else branch fetches reviewable lessons for which cards are available for review.

Reply:
@Ash-2k3 I'm not sure I follow. Didn't we have a discussion earlier about trying to deprecate the get_reviewable_translation_suggestions_by_offset(() method in the long run because its limit is unbounded and it can be given an unlimited number of exp_ids? My comment was addressing that this is one of the places that would need to be cleaned up (and refactored into a single-exploration-id flow) as part of that TODO.

The exp_ids are also already being passed in the elif branch so I'm also not sure what your second sentence means.

Reply:
Got it, thanks. I got confused earlier. Added a Todo Comment, thanks.